lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -483,10 +491,11 @@ pcmk_designated_controller(xmlNodePtr *xml, unsigned int message_timeout_ms)
  *
  * \return Standard Pacemaker return code
  *
- * \note This function returns \p EREMOTEIO if run on a Pacemaker Remote node
- *       with \p pacemaker-remoted running, since \p pacemakerd is not proxied
- *       to remote nodes. The fencer and CIB may still be accessible, but
- *       \p state will be \p pcmk_pacemakerd_state_invalid.
+ * \note This function sets \p state to \p pcmk_pacemakerd_state_running and
+ *       returns \p pcmk_rc_ok if the IPC connection attempt returns
+ *       \p EREMOTEIO. That code indicates that this is a Pacemaker Remote node
+ *       with \p pacemaker-remoted running. The node may be connected to the
+ *       cluster.
```

Comment:
This same change would need to be made for the public equivalent.

I'm ambivalent about it. It's technically breaking API compatibility since this return code was documented, but that's not a dealbreaker since this API is so new that I don't think it's gotten any external use yet. On the one hand, this is convenient for callers, but on the other hand, it's claiming something it doesn't know, and distinguishing pacemakerd vs pacemaker-remoted might one day be useful in some context.

Reply:
> this return code was documented

Kind of. The specific handling of `EREMOTEIO` was documented only for the internal function. The public doxygen just says "Standard Pacemaker return code" and says `"Get and output \p pacemakerd status"` (which is not quite the same as `pacemakerd-remote` status, granted). It doesn't give any details about when it will return which specific code.

> it's claiming something it doesn't know

It does know that `pacemaker-remoted` is running at least, which is the nearest meaningful analog we have for `pacemakerd` running on a Pacemaker Remote node. The hope of the doxygen note was to avoid promising something that isn't true.

This is a little bit of a hack, just so convenient. It's not a big deal to revert this though.

Reply:
Hmm, how would you feel about adding a `pcmk_pacemakerd_state_remote` enum value? It's not technically a `pacemakerd` state, but y'know... potentially useful indicator

I just remembered another reason why I did this: to allow `pcmk_cluster_queries.c:pacemakerd_event_cb()` to print something (besides "invalid") for the `pacemakerd` state.

Reply:
Ah I missed that EREMOTEIO isn't documented for the public API, that makes it less a problem. I like the idea of adding a new enum value so we're giving accurate information.